### PR TITLE
search: remove unused mode argument

### PR DESCRIPTION
**Pull request by rvantonder**
Stacked on https://github.com/sourcegraph/sourcegraph/pull/23943.

### CODE REVIEW

File: internal/search/zoekt/indexed_search.go [hunk 1]
```diff
@@ -335,7 +335,7 @@ func NewIndexedSearchRequest(ctx context.Context, args *search.TextParameters, t
 // have a repo: filter and consequently no rev: filter. This makes the code a bit
 // simpler because we don't have to resolve revisions before sending off (global)
 // requests to Zoekt.
-func zoektGlobalQuery(ctx context.Context, query zoektquery.Q, mode search.GlobalSearchMode, repoOptions search.RepoOptions, userPrivateRepos []types.RepoName) zoektquery.Q {
```
Comment (rvantonder) [inline internal/search/zoekt/indexed_search.go:338]: `mode` is no longer used here after previous changes--it's implied by the control flow.

File: internal/search/zoekt/indexed_search.go [hunk 2]
```diff
@@ -335,7 +335,7 @@ func NewIndexedSearchRequest(ctx context.Context, args *search.TextParameters, t
...
+func zoektGlobalQuery(ctx context.Context, query zoektquery.Q, repoOptions search.RepoOptions, userPrivateRepos []types.RepoName) zoektquery.Q {
```
Comment (stefanhengl) [inline internal/search/zoekt/indexed_search.go:338]: nit: remove `ctx` too?

**Review by stefanhengl — APPROVED** [review]

**Review by camdencheek — APPROVED** [review]

File: internal/search/zoekt/indexed_search.go (hunk 2 above)
Comment (rvantonder) [reply internal/search/zoekt/indexed_search.go:338]: heh, yes, missed that.

Comment (rvantonder) [reply internal/search/zoekt/indexed_search.go]: (good eye spotting this earlier too, where I missed it!)

**Pull request merged**